src/main/c/powerwaf_jni.c

```diff
@@ -469,6 +469,63 @@ JNIEXPORT jobjectArray JNICALL Java_io_sqreen_powerwaf_Powerwaf_getKnownAddresse
     return ret_jarr;
 }
 
+/*
+ * Class:     io_sqreen_powerwaf_Powerwaf
+ * Method:    getKnownActions
+ * Signature: (Lio/sqreen/powerwaf/PowerwafHandle;)[Ljava/lang/String;
+ */
+JNIEXPORT jobjectArray JNICALL Java_io_sqreen_powerwaf_Powerwaf_getKnownActions
+  (JNIEnv *env, jclass clazz, jobject handle_obj)
+{
+    UNUSED(clazz);
+
+    if (!_check_init(env)) {
+        return NULL;
+    }
+
+    ddwaf_handle nat_handle;
+    if (!(nat_handle = get_pwaf_handle_checked(env, handle_obj))) {
+        return NULL;
+    }
+
+    uint32_t size;
+    const char* const* actions = ddwaf_known_actions(nat_handle, &size);
+    if (!actions || size == 0 || size > INT_MAX /* jsize == int */) {
+        JAVA_LOG(DDWAF_LOG_DEBUG, "Found no actions in ruleset");
```

Comment:
This might be a bit misleading in case of an overflow `size > INT_MAX`, but I guess that this won't ever happen.